src/components/user/sign-up/index.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import {
+  Grid, Row, Col, Alert,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { signUpAPI } from '../../../store/actions/sign-up';
+import SignUpForm from './sign-up-form';
+
+class SignUp extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(fields) {
+    const { signUp } = this.props;
+    signUp(fields);
+  }
+
+  renderFormError() {
+    const { signUpUser } = this.props;
+    const { success, errorCode } = signUpUser;
+
+    let message;
+    if (!success) {
+      switch (errorCode) {
+        case 400:
+          message = 'One or more fields appears to be invalid. Please check the form and try again.';
+          break;
+        default:
+          message = 'An unknown error has occurred. Please try again.';
+          break;
+      }
+    }
+    return <Alert bsStyle="danger">{message}</Alert>;
+  }
+
+  render() {
+    const { signUpUser } = this.props;
+    const { success } = signUpUser;
+
+    if (success) {
+      return <Redirect to="./login" />;
+    }
+
+    return (
+      <div className="sign-up">
```

Comment:
You should prefix all top-level component classNames so they don't class with third party modules...

```
<div className="oc--sign-up">
  ...
</div>
```

Refactor all other custom components based on this comments. CSS will also have to be updated to match new classNames.

Reply:
what does the oc stand for? or did you mean to type os?

Reply:
Ah yes sorry I meant “os”